### DISCUSSION TITLE
Possible Jupyter Notebooks py5book/docs enhancements

### DISCUSSION THREAD

**Discussion by villares** [py5coding/py5generator#491] [Ideas]
Reading again the docs, I have a hunch that the first page on the  Jupyter Notebooks tutorials section (https://py5coding.org/tutorials/jupyter.html) should have a py5 kernel example and/or a plain (ipykernel?)  Python kernel example. 

Could it be the same or a variation of the example in the homepage (https://py5coding.org/#basic-example), maybe comparing it with a the plain Python kernel?

My point is, that it is mentioned that there are 2 special kernels (optional) and the next page is about the py5bot kernel (which is great), and then we have some other advanced uses... but I feel there is a gap... I realize that as we are using the py5book framework maybe it would better to break it in more pages (one for each kernel?), but I think I'd like them best in the first page.

Could we have something like:

- Using Jupyter Notebooks https://py5coding.org/tutorials/jupyter.html
    - " [Project Jupyter](https://jupyter.org/) is a collection of open source projects providing services and tools for interactive computing across many programming languages [...] Built into py5 are many features to make it compatible with Jupyter Notebooks. This empowers py5 to benefit from Jupyter’s strengths. This section provides some tutorials that highlight py5’s Jupyter-related features and how to make best use of them.
It is recommended that you install both of py5’s Jupyter kernels, **py5 and pyt5bot**. Instructions for how to do so are found at the bottom of the [py5 install page](https://py5coding.org/content/install.html)."

     - Plain Python kernel basic example
      
  ```python
  import py5

  def setup():
      py5.size(400, 400)
      py5.rect_mode(py5.CENTER)
  
  
  def draw():
      py5.square(py5.mouse_x, py5.mouse_y, 10)
  
  
  def mouse_clicked():
      fill(py5.random_int(255), py5.random_int(255), py5.random_int(255))
  
  
  py5.run_sketch()
  ``` 
      
  - py5 kernel basic example
      
  ```python
  def setup():
      size(400, 400)
      rect_mode(CENTER)
  
  
  def draw():
      square(mouse_x, mouse_y, 10)
  
  
  def mouse_clicked():
      fill(random_int(255), random_int(255), random_int(255))
  
  
  run_sketch()
  ``` 
      
- Introducing py5bot https://py5coding.org/tutorials/jupyter_introducing_py5bot.html

**Comment by hx2A** [comment]
The Jupyter notebook documentation could be improved and this is a good way to make it more clear about the difference between the default Python kernel and the py5 kernel. Would like to make a PR for this? The file you'd have to change is:

https://github.com/py5coding/py5book/blob/main/tutorials/jupyter.ipynb

Note that there is also a jupyter.md file in that same directory. This is setup to sync the two files so you can edit either one and see the change in both. You might have to do something on your machine for the syncing to work though. Installing jupytext might be enough:

https://jupytext.readthedocs.io/en/latest/paired-notebooks.html

If you edit the jupyter.ipynb file and don't see the other file change, don't worry about it, I can take care of it for you. The notebook syncing thing is very useful though.

**Reply by villares** [reply]
Cool! I'm doing a `conda install jupytext -c conda-forge` on my `py5conding` conda env and let's see how it goes :)
